knowledge-base/grid-show-tooltip-on-column-header.md

```diff
@@ -0,0 +1,109 @@
+2. Monitor the column width changes by utilizing the [Grid State](slug:components/grid/state) and its [ColumnState](slug:components/grid/state#information-in-the-grid-state) property.
+3. Implement the [TelerikTooltip](slug:components/tooltip/overview) component and utilize its [Template](slug:components/tooltip/template) to define the content of the tooltip, which should match the full column header text.
+4. Apply a custom CSS class to the column header content when the width of the column is reduced enough to hide its full content. This class will be used as the target selector for the TelerikTooltip.
+
+>caption Show TelerikTooltip on the Grid column header
+
+````RAZOR
+<strong>Resize first column to see the result</strong>
+<br/>
+<TelerikGrid Data="@MyData" Resizable="true"
+             OnStateChanged="@((GridStateEventArgs<SampleData> args) => HandleColumnWidthChange(args))"
+             @ref="GridRef"
+             Width="300px">
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Name))"
+                    Width="165px">
+            <HeaderTemplate>
+                <span class="@HeaderClass">
+                    Long Employee Name
+                </span>
+            </HeaderTemplate>
+        </GridColumn>
+        <GridColumn Field="@(nameof(SampleData.Team))" />
+    </GridColumns>
+</TelerikGrid>
+
+<TelerikTooltip TargetSelector=".employee-header"
+                Class="my-callout">
+    <Template>
+        Long Employee Name
+    </Template>
+</TelerikTooltip>
+
+<style>
+    .my-callout .k-callout {
+        left: 30px !important;
+    }
+</style>
+
+@code {
+    private string HeaderClass { get; set; } = string.Empty;
+    private string HeaderId { get; set; } = string.Empty;
+    private TelerikGrid<SampleData> GridRef { get; set; } = null!;
+
+    private IEnumerable<SampleData> MyData = Enumerable.Range(1, 10).Select(x => new SampleData
+        {
+            Id = x,
+            Name = "name " + x,
+            Team = "team" + x
+        });
+
+    private async Task HandleColumnWidthChange(GridStateEventArgs<SampleData> args)
+    {
+        var employeeColumnWidth = args.GridState.ColumnStates.First(c => c.Index == 1).Width.Replace("px", "");
+        var columnWidth = double.Parse(employeeColumnWidth);
+
+        if (columnWidth < 160)
+        {
+            HeaderClass = "employee-header";
+        }
+        else
+        {
+            HeaderClass = string.Empty;
+        }
+
+        await GridRef.SetStateAsync(args.GridState);
+    }
+
+    public class SampleData
+    {
+        public int Id { get; set; }
+        public string Name { get; set; } = string.Empty;
+        public string Team { get; set; } = string.Empty;
+    }
+}
+````
+
+The additional CSS is used to adjust the position of the tooltip callout. Modify this CSS based on your application's specific layout and design requirements.
+
+## See Also
```

Comment:
Cross-link with https://www.telerik.com/blazor-ui/documentation/knowledge-base/tooltip-in-grid